amazon-connect/callstats-amazon-connect-shim.js

```diff
@@ -1,6 +1,92 @@
 /*! callstats Amazon SHIM version = 1.1.8 */
 
 (function (global) {
+  class VoiceActivityDetection {
+    constructor(stream, callback) {  
+      this.AudioContext = window.AudioContext || window.webkitAudioContext;
+      this.audioContext = new this.AudioContext();
+      this.analyser = this.audioContext.createAnalyser();
+      this.analyser.fftSize = 512;
+      this.analyser.smoothingTimeConstant = 0.1;
+      this.fftBins = new Float32Array(this.analyser.frequencyBinCount);
+      this.mediaStreamSource = this.audioContext.createMediaStreamSource(stream);
+      this.mediaStreamSource.connect(this.analyser);
+      this.isSpeaking = false;
+      this.maxVolumeHistory = [];
+      for (var i=0; i < 10; i++) {
+        this.maxVolumeHistory.push(0);
+      }
+      this.threshold = -50;
```

Comment:
Can we add comments about these parameters here?

Reply:
Where is the -50 coming from? Are all values from FFT negative here?

Reply:
Conversation is considered to be between 40-60dB, so I wonder if it 50 and results of FFT are negative?

Reply:
webAudio API defines AnalyserNode,

It has the following attributes
maxDecibels, of type double
maxDecibels is the maximum power value in the scaling range for the FFT analysis data for conversion to unsigned byte values. The default value is -30. If the value of this attribute is set to a value less than or equal to minDecibels, an IndexSizeError exception MUST be thrown.
minDecibels, of type double
minDecibels is the minimum power value in the scaling range for the FFT analysis data for conversion to unsigned byte values. The default value is -100. If the value of this attribute is set to a value more than or equal to maxDecibels, an IndexSizeError exception MUST be thrown.

So the default range is -100 to -30
0 dB is the loudest possible sound, -10 dB is a 10th of that

-50 is the threshold if the speech is present in the stream.